### ISSUE TITLE
Reporting exception handling doesn't work in Jupyter

### ISSUE THREAD

**Issue by Jihoon** [iiasa/ixmp#206]
When certain entries are empty for Reporter.get() call, now it complains:
`AttributeError: 'NoneType' object has no attribute '__traceback__'`

Better to give out a warning message after checking the entries. No way users can understand what's going on.

## As described in duplicate #208

As reported by @Jihoon: in Jupyter kernels, the ixmp.reporting exception-handling code produces unhelpful tracebacks like the following:
```
rep.get(rep.full_key('emission_factor'))

Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py", line 3343, in run_code
    self.showtraceback(running_compiled_code=True)
  File "C:\ProgramData\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py", line 2045, in showtraceback
    self._showtraceback(etype, value, stb)
  File "C:\ProgramData\Anaconda3\lib\site-packages\ipykernel\zmqshell.py", line 546, in _showtraceback
    u'evalue' : py3compat.safe_unicode(evalue),
  File "C:\ProgramData\Anaconda3\lib\site-packages\ipython_genutils\py3compat.py", line 65, in safe_unicode
    return unicode_type(e)
  File "h:\mydocuments\message\ixmp\ixmp\reporting\__init__.py", line 667, in __str__
    tb = TracebackException.from_exception(self.cause, capture_locals=True)
  File "C:\ProgramData\Anaconda3\lib\traceback.py", line 526, in from_exception
    return cls(type(exc), exc, exc.__traceback__, *args, **kwargs)

AttributeError: 'NoneType' object has no attribute '__traceback__'
```

This is contrary to its goal of providing intelligible messages when `.get()` fails.

**Comment by khaeru** [comment]
Thanks for reporting this!

To state the issue in technical terms:
- At the request of @OFR-IIASA, in #188 I added `ixmp.reporting.ComputationError` class: https://github.com/iiasa/ixmp/blob/e6b3c0d663b97b09a6ccec331ede1e2def992ca6/ixmp/reporting/__init__.py#L641-L700
- The purpose:
  - When `dask` (the Python package that executes the reporting graph) encounters an error in one operation, it prints a very long traceback, including lots of dask internal calls. This is confusing and makes debugging hard.
  - The `ComputationError` class strips out that dask-related information, leaving only (a) the user code and (b) `ixmp`/`ixmp.Reporting` internals.
  - This more clearly points to the reporting operation that is failing.
- However, this was developed/tested only for **non-interactive use**, i.e. inside of Python scripts executed like `$ python script.py`.
- @Jihoon's use-case (and perhaps others') is **interactive** use, through an editor like Spyder that provides an interactive experience (called a “[REPL](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop)”).
  - Jupyter notebooks also implement a REPL.
- These tools implement their *own* custom exception handling.
- That exception handling clashes with the code in `ComputationError`.
- The result is the unhelpful exception messages reported—the opposite of what `ComputationError` is supposed to achieve.

To develop a fix, someone should:
- Create a snippet Jupyter notebook that creates a Reporter and uses it to generate an error.
- Test that the error gives the text returned by `ComputationError`. Currently, this test should fail.
- Improve `ComputationError` so that it passes both this new test *and* the existing tests, e.g. https://github.com/iiasa/ixmp/blob/e6b3c0d663b97b09a6ccec331ede1e2def992ca6/tests/test_reporting.py#L423-L425

**Comment by khaeru** [comment]
Sorry for the confusion—I think I had suggested the title for this issue in relation to a _different_ issue that we were discussing around the same time. Then the two got mixed up.

I've created #248 to describe the specific issue with filters that are too restrictive. That is _one possible cause_ of a ComputationErrors. This issue (with new title taken from #208) deals with how _all_ ComputationErrors are handled.
